ocaml/parsing/parsetree.mli

```diff
@@ -921,16 +921,24 @@ and module_substitution =
 and module_type_declaration =
     {
      pmtd_name: string loc;
-     pmtd_type: module_type option;
+     pmtd_type: module_type_declaration_type;
      pmtd_attributes: attributes;  (** [... [\@\@id1] [\@\@id2]] *)
      pmtd_loc: Location.t;
     }
 (** Values of type [module_type_declaration] represents:
    - [S = MT],
+     when {{!module_type_declaration.pmtd_type}[pmtd_type]} is [Pmtd_define MT].
    - [S] for abstract module type declaration,
-     when {{!module_type_declaration.pmtd_type}[pmtd_type]} is [None].
+     when {{!module_type_declaration.pmtd_type}[pmtd_type]} is [Pmtd_abstract].
+   - [S = _] for inferring module type from definition in the signature,
```

Comment:
Might be simpler if we just say "inferring module type". Because if we say "in the signature" it involves the distinction between signature and structure, and the fact that underscore is invalid in signature, etc.. But this is `parsetree.mli` whose `module_type_decalration` doesn't know whether it's in a signature or a structure.